tests/components/pooldose/test_config_flow.py

```diff
@@ -237,3 +238,40 @@ async def test_duplicate_entry_aborts(
     )
     assert result["type"] is FlowResultType.ABORT
     assert result["reason"] == "already_configured"
+
+
+async def test_dhcp_flow(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test the full DHCP config flow."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_DHCP},
+        data=DhcpServiceInfo(
+            ip="192.168.0.123", hostname="kommspot", macaddress="a4e57caabbcc"
+        ),
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "dhcp_confirm"
+
+    result = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == "PoolDose TEST123456789"
+    assert result["data"] == {CONF_HOST: "192.168.0.123"}
+    assert result["result"].unique_id == "TEST123456789"
+
+
+async def test_dhcp_no_serial_number(
+    hass: HomeAssistant, mock_pooldose_client: AsyncMock, mock_setup_entry: AsyncMock
+) -> None:
+    """Test that the DHCP flow aborts if no serial number is found."""
+    mock_pooldose_client.device_info = {"NAME": "Pool Device", "MODEL": "POOL DOSE"}
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_DHCP},
+        data=DhcpServiceInfo(
+            ip="192.168.0.123", hostname="kommspot", macaddress="a4e57caabbcc"
+        ),
+    )
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "no_serial_number"
```

Comment:
We should test all the ways this can return an abort. We should also have atest where we try to set up the same device twice.